### PR TITLE
Improvement/ker/phase reset drop

### PR DESCRIPTION
**Pull request by keriksson-rosenqvist**
Clean up instructions added by `builder.measure()`. This no longer adds a `PhaseReset` at the end and only syncs the measured `Qubit`s' channels at the start, not all entangled devices.

### CODE REVIEW

**Comment by keriksson-rosenqvist** [comment]
@bryn-bell Here is the PR I mentioned. Let me know how you want to proceed.

File: src/qat/purr/compiler/instructions.py [hunk 1]
```diff
@@ -869,9 +869,7 @@ def instructions(self) -> List[QuantumInstruction]:
         instructions = [Synchronize(list(self._entangled_qubits))]
```
Comment (willh727-OQC) [inline src/qat/purr/compiler/instructions.py:869]: Is it also worth changing this to targets? It seems to be part of the confusion

**Review by willh727-OQC — APPROVED** [review]
We should also think about stripping back the complex of measure, it seems to be producing a lot of unexpected behavior

**Review by hamidelmaazouz — APPROVED** [review]

File: src/qat/purr/compiler/instructions.py (hunk 1 above)
Comment (keriksson-rosenqvist) [reply src/qat/purr/compiler/instructions.py:869]: This will expand to all `Qubit`s measured in the `MeasureBlock`. Does not care about entanglement.

Comment (willh727-OQC) [reply src/qat/purr/compiler/instructions.py:869]: I ment measurement targets not the entangled_qubits! So the conclusions is to change this to measurement targets from chats.

Comment (keriksson-rosenqvist) [reply src/qat/purr/compiler/instructions.py]: Done as of last commit.

**Review by lcauser-oqc — APPROVED** [review]
LGTM!

**Pull request merged**